### PR TITLE
Created datamodels and CRUD operations for all session endpoints

### PR DESCRIPTION
**Pull request by tlulu**

### CODE REVIEW

File: src/main/scala/com/shuttleql/services/session/ClubSession.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.shuttleql.services.session
+
+/* Singleton class for a club session
+ */
+object ClubSession {
+	private var sessionId: Long = -1
```
Comment (c9dong) [inline src/main/scala/com/shuttleql/services/session/ClubSession.scala]: can you wrap -1 to a variable named INVALID_SESSION?

Comment (c9dong) [reply src/main/scala/com/shuttleql/services/session/ClubSession.scala]: nvm, current session should be in database with an active flag

**Review by c9dong — COMMENTED** [review]
Almost flawless

File: src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    val db = initDb
+    val now = new java.sql.Timestamp(System.currentTimeMillis())
+
+    try {
+      val userToSession = Await.result(db.run(this.filter(_.sessionId === sessionId)
+            .filter(_.userId === userId).result).map(_.headOption), Duration.Inf)
+
+      Await.result(db.run(this.filter(_.sessionId === sessionId)
+            .filter(_.userId === userId)
+            .map(x => (x.checkedOutAt))
```
Comment (jasonf7) [inline src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala:53]: `this.filter(_.sessionId === sessionId && _.userId === userId)`
`.map(_.checkedOutAt)`

File: src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+  }
+
+  def update(userId: Long, sessionId: Long): Option[UserToSession] = {
+    val db = initDb
+    val now = new java.sql.Timestamp(System.currentTimeMillis())
+
+    try {
+      val userToSession = Await.result(db.run(this.filter(_.sessionId === sessionId)
+            .filter(_.userId === userId).result).map(_.headOption), Duration.Inf)
```
Comment (jasonf7) [inline src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala:49]: `this.filter(_.sessionId === sessionId && _.userId === userId)`

Comment (jasonf7) [reply src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala:49]: can you align transformations on new line like u did below

**Review by jasonf7 — CHANGES_REQUESTED** [review]

File: src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala (hunk 3 above)
Comment (tlulu) [reply src/main/scala/com/shuttleql/services/session/UserToSessionsDAO.scala:49]: no

File: src/main/scala/com/shuttleql/services/session/SessionServiceServlet.scala [hunk 4]
```diff
@@ -4,50 +4,116 @@ import org.scalatra.json._
...
+        InternalServerError(reason = "Error with creating club session.")
+    }
   }
 
   put("/end") {
-  	// check for secret key
-  	// ClubSession.end
-    logger.info("end")
+  	SessionsDAO.end()
   	Ok()
```
Comment (jasonf7) [inline src/main/scala/com/shuttleql/services/session/SessionServiceServlet.scala]: !!!!!!!!! NoContent

**Review by jasonf7 — CHANGES_REQUESTED** [review]

**Pull request merged**